src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -52,8 +57,8 @@ public EventPublishingController(final EventPublisher publisher,
                                      final EventTypeMetricRegistry eventTypeMetricRegistry,
                                      final BlacklistService blacklistService,
                                      final NakadiKpiPublisher nakadiKpiPublisher,
-                                     @Value("${nakadi.kpi.event-types.nakadiBatchPublished}")
-                                         final String kpiBatchPublishedEventType) {
+                                     @Value("${nakadi.kpi.event-types.nakadiBatchPublished}") final
```

Comment:
I wonder if wouldn't it be possible to capture this request via a springboot filter and if you further need to add things to the `span` that are not available in the filter, you could still create the span in the filter and pass it to the controller as the last parameter. Something like the client resolver https://github.com/zalando/nakadi/blob/1643b2182c1b4bb257ad6d822f48971eeaf9f91f/src/main/java/org/zalando/nakadi/config/WebConfig.java#L85 if I'm not mistaken.

Also, how is this code coexisting with already configured operation to track publishing? Did it replace that? I talk about the one in https://github.com/zalando/nakadi/blob/41fd11ada6d513cf5c4fe8e9fb16e82c12cf326a/src/main/java/org/zalando/nakadi/filters/LoggingFilter.java#L185